miden-lib/build.rs

```diff
@@ -37,6 +37,12 @@ fn main() -> Result<()> {
     println!("cargo:rerun-if-changed=asm");
     println!("cargo:rerun-if-changed={KERNEL_V0_RS_FILE}");
 
+    // The docs.rs build pipeline has a read-only filesystem, so we want to return early or
+    // otherwise the docs will fail to build there.
+    if env::var("DOCS_RS").is_ok() {
+        return Ok(());
+    }
+
```

Comment:
If we remove this, then the docs build succeeds (tested it locally).

This is fine since this build script now no longer writes to `src` but only `OUT_DIR`.

Reply:
Resolving this as the approach changed.